mod/beacon/validator/propose.go

```diff
@@ -298,3 +248,63 @@ func (s *Service[
 	}
 	return envelope, nil
 }
+
+// BuildBlockBody assembles the block body with necessary components.
+func (s *Service[
+	BeaconBlockT, BeaconBlockBodyT, BeaconStateT,
+	BlobSidecarsT, DepositStoreT, ForkDataT,
+]) buildBlockBody(
+	ctx context.Context,
+	st BeaconStateT,
+	blk BeaconBlockT,
+	reveal crypto.BLSSignature,
+	envelope engineprimitives.BuiltExecutionPayloadEnv[*types.ExecutionPayload],
+) error {
+	// Assemble a new block with the payload.
+	body := blk.GetBody()
+	if body.IsNil() {
+		return ErrNilBlkBody
+	}
+
+	// Set the reveal on the block body.
+	body.SetRandaoReveal(reveal)
+
+	// If we get returned a nil blobs bundle, we should return an error.
+	blobsBundle := envelope.GetBlobsBundle()
+	if blobsBundle == nil {
+		return ErrNilBlobsBundle
+	}
+
+	// Set the KZG commitments on the block body.
+	body.SetBlobKzgCommitments(blobsBundle.GetCommitments())
+
+	depositIndex, err := st.GetEth1DepositIndex()
+	if err != nil {
+		return ErrNilDepositIndexStart
+	}
+
+	// Dequeue deposits from the state.
+	deposits, err := s.bsb.DepositStore(ctx).GetDepositsByIndex(
+		depositIndex,
+		s.chainSpec.MaxDepositsPerBlock(),
+	)
+	if err != nil {
+		return err
+	}
+
+	// Set the deposits on the block body.
+	body.SetDeposits(deposits)
+
+	// TODO: assemble real eth1data.
+	body.SetEth1Data(&types.Eth1Data{
+		DepositRoot:  primitives.Bytes32{},
+		DepositCount: 0,
+		BlockHash:    common.ZeroHash,
+	})
+
+	if err = body.SetExecutionData(envelope.GetExecutionPayload()); err != nil {
```

Comment:
can just return this directly

Reply:
@itsdevbear tks a lot for your time and review. Updated!